src/jp2image.cpp

```diff
@@ -198,6 +194,16 @@ namespace Exiv2
         return result;
     }
 
+static void boxes_check(size_t b,size_t m)
```

Comment:
This check seems a little arbitrary to me, maybe we should find out why there's an infinite loop in the first place?

Reply:
I agree in which it seems a bit arbitraty (or at least we do not have a reference to know about the maximum number of boxes). My approach would be to try to merge this as it is, and create another issue to investigate about this in the future. 

I was searching the JPEG 2000 specification, but it seems it is not publicly available :cry: 
https://www.iso.org/standard/37674.html